File: argon-plugins-tube/src/main/scala/dev/argon/plugins/tube/TubeLibOptions.scala
package dev.argon.plugins.tube

import dev.argon.compiler.Context
import dev.argon.io.ZipFileResource
import dev.argon.options.OptionDecoder

final case class TubeLibOptions[-R, +E, ContextOptions]
(
  tube: ZipFileResource[R, E],
)


object TubeLibOptions:

  given optionDecoder[R, E >: TubeError, ContextOptions](using OptionDecoder[R, E, ContextOptions]): OptionDecoder[R, E, TubeLibOptions[R, E, ContextOptions]] =
    OptionDecoder.derive

end TubeLibOptions

File: plugin/tube/java/src/main/java/dev/argon/plugins/tube/TubeLibOptions.java
package dev.argon.plugins.tube;

import dev.argon.plugin.api.BinaryResource;
import dev.argon.plugin.api.options.Options;
import dev.argon.plugin.api.options.OptionsDecoding;

@Options(decoding = OptionsDecoding.NORMAL)
public record TubeLibOptions<E extends Throwable>(
   BinaryResource<E> tube
) {}